tests/test-cmd-upload-texture.cpp

```diff
@@ -1,247 +1,272 @@
 
-        TextureDesc desc = srcTexture->getDesc();
+            TextureData newData;
+            newData.init(currentData.device, currentData.desc, TextureInitMode::Random, 1000);
 
-        uint32_t layerCount = desc.getLayerCount();
+            // Create command encoder
+            auto device = c->getDevice();
+            auto queue = device->getQueue(QueueType::Graphics);
+            auto commandEncoder = queue->createCommandEncoder();
 
-        auto queue = device->getQueue(QueueType::Graphics);
-        auto commandEncoder = queue->createCommandEncoder();
+            uint32_t layerCount = currentData.desc.getLayerCount();
 
-        commandEncoder->uploadTextureData(
-            srcTexture,
-            {0, desc.mipLevelCount, 0, layerCount},
-            {0, 0, 0},
-            Extents::kWholeTexture,
-            srcTextureInfo->subresourceDatas.data(),
-            srcTextureInfo->subresourceDatas.size()
-        );
+            // Replace alternate layers
+            for (uint32_t layer = 0; layer < layerCount; layer++)
+            {
+                if ((layer % 2) == 1)
+                {
+                    // Copy the subresource data from inverted layer index to this layer
+                    auto srdata = newData.getLayerFirstSubresourceData(layer);
+                    commandEncoder->uploadTextureData(
+                        c->getTexture(),
+                        {0, newData.desc.mipLevelCount, layer, 1},
+                        {0, 0, 0},
+                        Extents::kWholeTexture,
+                        srdata,
+                        currentData.desc.mipLevelCount
+                    );
+                }
+            }
 
-        queue->submit(commandEncoder->finish());
-        queue->waitOnHost();
+            // Execute all operations
+            queue->submit(commandEncoder->finish());
+            queue->waitOnHost();
 
-        for (uint32_t layerIndex = 0; layerIndex < layerCount; layerIndex++)
-        {
-            for (uint32_t mipLevel = 0; mipLevel < desc.mipLevelCount; mipLevel++)
+            // Verify alternate layers from original and new data
+            for (uint32_t layer = 0; layer < layerCount; layer++)
             {
-                uint32_t subresourceIndex = getSubresourceIndex(mipLevel, desc.mipLevelCount, layerIndex);
+                if ((layer % 2) == 0)
+                    currentData.checkLayersEqual(c->getTexture(), layer, layer);
+                else
+                    newData.checkLayersEqual(c->getTexture(), layer, layer);
+            }
+        }
+    );
+}
 
-                SubresourceLayout layout;
-                REQUIRE_CALL(srcTexture->getSubresourceLayout(mipLevel, &layout));
+GPU_TEST_CASE("cmd-upload-texture-single-mip", D3D12 | Vulkan | WGPU)
+{
+    TextureTestOptions options(device);
+    options.addVariants(TTShape::All, TTArray::Off, TTMip::On, TextureInitMode::Random, TTFmtDepth::Off);
```

Comment:
we're not testing updating a single mip level in a layered texture .. maybe overkill, just saying